### PR TITLE
fix: add realm_id to CombatOutcome and OrderAccepted + smart_contract types

### LINKED ISSUE #33: bug: `OrderAccepted` and `CombatOutcome` events should include realm_id

### PR DESCRIPTION
**Pull request by edisontim**
Closes #33

### CODE REVIEW

**Comment by edisontim** [comment]
Need to add one test that check if a `CombatOutcome` is emitted well when the attacker loses the attack

File: contracts/src/models/npc.cairo [hunk 1]
```diff
@@ -47,63 +47,36 @@ fn random_role(source: u128) -> u8 {
...
 struct Characteristics {
     age: u8,
-    hunger: u8,
-    happiness: u8,
-    belligerency: u8,
     role: u8,
     sex: u8,
 }
 
 fn random_characteristics(randomness: @Span<u128>) -> Characteristics {
```
Comment (cwastche) [inline contracts/src/models/npc.cairo:54]: Do we still need this function if we generate sex and role within the GPT prompt ?

File: contracts/src/systems/npc/contracts.cairo [hunk 2]
```diff
@@ -1,49 +1,63 @@
...
+                    name: old_npc.name
+                })
+            );
         }
     }
 
     #[external(v0)]
     impl NpcImpl of INpc<ContractState> {
-        fn spawn_npc(self: @ContractState, world: IWorldDispatcher, realm_id: felt252) -> felt252 {
+        fn spawn_npc(self: @ContractState, world: IWorldDispatcher, realm_id: u128, character_trait: felt252, name: felt252) -> u128 {
```
Comment (cwastche) [inline contracts/src/systems/npc/contracts.cairo:60]: Same comment as above, should we pass sex + role just like the trait, as we would get them from the prompt ?

File: contracts/src/systems/trade/contracts/trade_systems.cairo [hunk 3]
```diff
@@ -314,17 +316,20 @@ mod trade_systems {
...
                 trade_id, 
-                maker_id: trade.maker_id,
-                taker_id, 
-                maker_resources: resource_chest::get_contents(world, trade.maker_resource_chest_id),
-                taker_resources: resource_chest::get_contents(world, trade.taker_resource_chest_id),
+                maker_realm_entity_id: maker_realm.entity_id,
+                maker_realm_id: maker_realm.realm_id,
+                taker_realm_entity_id: taker_realm.entity_id,
+                taker_realm_id: taker_realm.realm_id,
+                    maker_resources: resource_chest::get_contents(world, trade.taker_resource_chest_id),
```
Comment (cwastche) [inline contracts/src/systems/trade/contracts/trade_systems.cairo:328]: Sanity check, maker takes the content of the chest created by taker and vice versa. Correct ?

**Review by cwastche — APPROVED** [review]
lgtm all in all, just some small comments on the npc_spawn characteristics.

File: contracts/src/systems/trade/contracts/trade_systems.cairo (hunk 3 above)
Comment (edisontim) [reply contracts/src/systems/trade/contracts/trade_systems.cairo:328]: Yes exactly, we want to have the resources that the maker/taker is giving. But the content of the chests are the resources that each of them is getting

File: contracts/src/systems/npc/contracts.cairo (hunk 2 above)
Comment (edisontim) [reply contracts/src/systems/npc/contracts.cairo]: Changed

File: contracts/src/models/npc.cairo [hunk 4]
```diff
@@ -47,63 +47,36 @@ fn random_role(source: u128) -> u8 {
...
-    Characteristics {
-        age: 1,
-        hunger: (*randomness[1] % 10).try_into().unwrap(),
-        happiness: (u128_wrapping_sub(*randomness[1], 3) % 10).try_into().unwrap(),
-        belligerency: (u128_wrapping_sub(*randomness[1], 16) % 10).try_into().unwrap(),
-        role: random_role(*randomness[0]),
-        sex: random_sex(*randomness[2]),
-    }
+    Characteristics { age: 1, role: random_role(*randomness[0]), sex: random_sex(*randomness[1]), }
```
[the remark this answers is not in the document]
Comment (edisontim) [reply contracts/src/models/npc.cairo]: Changed

**Review by cwastche — APPROVED** [review]
lgtm

**Pull request merged**